### ISSUE TITLE
Support also `%__patch` macro

### ISSUE THREAD

**Issue by jankaluza** [rpminspect/rpminspect#1284]
**Describe the bug**

Some packages (at least the sssd - https://gitlab.com/redhat/centos-stream/rpms/sssd/-/blob/c8s/sssd.spec?ref_type=heads#L568) uses following code to apply patches:

```
for p in %patches ; do
    %__patch -p1 -i $p
    UpdateTimestamps -p1 $p
done
```

This is so fer unsupported by the rpminspect and it thinks the patches are not applied.

**To Reproduce**

Run rpminspect on https://kojihub.stream.rdu2.redhat.com/koji/buildinfo?buildID=37112

**Expected behavior**

The patches are not reported as not applied.

**Environment:**
 - Fedora 38.

**Comment by dcantrell** [comment]
When I run rpminspect against the SRPM at https://kojihub.stream.rdu2.redhat.com/koji/buildinfo?buildID=37112, the patches inspection passes.  The SRPM has no patches in it and none are defined in the spec file.

The %prep block does carry the loop over %patches to apply and then call UpdateTimestamps.  Looks like this is just there in case they add patches to the package.

I don't understand why UpdateTimestamps needs to exist.  The comment indicates it's for mismatches with the .pyo or .pyc files.  I would say a better approach would be to drop the .pyo and .pyc files that are in the source archive anyway.  Do that after applying any patches using %autopatch
